src/libmongoc/src/mongoc/mongoc-client-side-encryption.c

```diff
@@ -1181,6 +1198,39 @@ _parse_extra (const bson_t *extra,
             GOTO (fail);
          }
       }
+
+      if (bson_iter_init_find (&iter, extra, "csflePath")) {
+         if (!BSON_ITER_HOLDS_UTF8 (&iter)) {
+            bson_set_error (error,
+                            MONGOC_ERROR_CLIENT,
+                            MONGOC_ERROR_CLIENT_INVALID_ENCRYPTION_ARG,
+                            "Expected a string for 'csflePath'");
+            GOTO (fail);
+         }
+         size_t len;
+         const char *ptr = bson_iter_utf8_unsafe (&iter, &len);
+         bson_free (topology->csfle_override_path);
+         topology->csfle_override_path = bson_strdup (ptr);
+      }
+
+      if (bson_iter_init_find (&iter, extra, "csfleRequired")) {
+         if (!BSON_ITER_HOLDS_BOOL (&iter)) {
+            bson_set_error (error,
+                            MONGOC_ERROR_CLIENT,
+                            MONGOC_ERROR_CLIENT_INVALID_ENCRYPTION_ARG,
+                            "Expected a bool for 'csfleRequired'");
+            GOTO (fail);
+         }
+         topology->csfle_required = bson_iter_bool_unsafe (&iter);
+      }
+
+      /** The __csfleDisabled option is a private API option only used for
+       * testing purposes. */
+      if (bson_iter_init_find (&iter, extra, "__csfleDisabled")) {
+         BSON_ASSERT (BSON_ITER_HOLDS_BOOL (&iter) &&
+                      "__csfleDisabled is for testing purposes only");
```

Comment:
This error message seems a little oddly worded given the assertion being tested. It seems to suggest `__csfleDisabled` shouldn't be present at all, rather than that it should contain a `bool` value. Should it be worded something like, "__csfleDisabled should only have been defined by X for testing purposes"?

Reply:
Probably, yes